ckan/lib/io.py

```diff
@@ -53,3 +58,22 @@ def decode_path(p: Union[bytes, Any]) -> str:
     if not isinstance(p, bytes):
         raise TypeError(u'Can only decode bytes, not {}'.format(type(p)))
     return p.decode(_FILESYSTEM_ENCODING)
+
+
+def get_ckan_temp_directory() -> str:
+    """
+    Returns the path to a securely created temporary directory that can be used
+    to store internal generated files like webassets, i18n files, etc
+
+    It is used as fallback when `ckan.storage_path` is not defined.
+    """
+
+    unique_suffix = hashlib.sha256(config["SECRET_KEY"].encode()).hexdigest()[:10]
```

Comment:
@smotornyuk looks like "SECRET_KEY" is our one secret to rule them all, re #7651

Reply:
Yes, I think we should work towards that: https://github.com/ckan/ckan/issues/7409